llvm/lib/Target/X86/X86InstrInfo.cpp

```diff
@@ -2355,29 +2355,30 @@ MachineInstr *X86InstrInfo::commuteInstructionImpl(MachineInstr &MI, bool NewMI,
     // If we're optimizing for size, try to use MOVSD/MOVSS.
     if (MI.getParent()->getParent()->getFunction().hasOptSize()) {
       unsigned Mask;
+      unsigned NewOpc;
       switch (Opc) {
       default:
         llvm_unreachable("Unreachable!");
       case X86::BLENDPDrri:
-        Opc = X86::MOVSDrr;
+        NewOpc = X86::MOVSDrr;
         Mask = 0x03;
         break;
       case X86::BLENDPSrri:
-        Opc = X86::MOVSSrr;
+        NewOpc = X86::MOVSSrr;
         Mask = 0x0F;
         break;
       case X86::VBLENDPDrri:
-        Opc = X86::VMOVSDrr;
+        NewOpc = X86::VMOVSDrr;
         Mask = 0x03;
         break;
       case X86::VBLENDPSrri:
-        Opc = X86::VMOVSSrr;
+        NewOpc = X86::VMOVSSrr;
         Mask = 0x0F;
         break;
       }
       if ((MI.getOperand(3).getImm() ^ Mask) == 1) {
         WorkingMI = CloneIfNew(MI);
-        WorkingMI->setDesc(get(Opc));
+        WorkingMI->setDesc(get(NewOpc));
```

Comment:
I don't understand why it can solve the problem. There's no difference between `Opc` and `NewOpc`, except `NewOpc` exposes more risk that may be used without assigned a value.

Reply:
I updated the description, there is a `fallthrough` at line 2386, and we check the old opcode again.

Reply:
But the `break` at line 2383 won't make it jumps to `fallthrough`

Reply:
Should it be
```
if ((MI.getOperand(3).getImm() ^ Mask) == 1) {
  switch (Opc) {
  ...
```

Reply:
> But the `break` at line 2383 won't make it jumps to `fallthrough`

Yes. The problem is that if `MI.getOperand(3).getImm() ^ Mask) == 1` is `false`, the fallthrough will happen.

`break` at line 2383 is expected b/c `WorkingMI` is determined.

Reply:
> Should it be
> 
> ```
> if ((MI.getOperand(3).getImm() ^ Mask) == 1) {
>   switch (Opc) {
>   ...
> ```

I think no. We don't need to distinguish the above 4 opcodes for this case.

Reply:
I mean the `opc` change only valid if `(MI.getOperand(3).getImm() ^ Mask) == 1`, we may refactor like:
```
unsigned Mask = 0;
if (Opc == X86::BLENDPDrri || Opc == X86::VBLENDPDrri)
  Mask = 0x03;
else if (Opc == X86::BLENDPSrri || Opc == X86::VBLENDPSrri)
  Mask = 0x0F;
else
  llvm_unreachable("Unreachable!");
if ((MI.getOperand(3).getImm() ^ Mask) == 1) {
  switch (Opc) {
  ..
}
```

Reply:
Good suggestion! Done.